### PR TITLE
Tdd

### PR DESCRIPTION
**Pull request by gasangw**
- Create a class called Solver.
- Create a method called factorial that takes one argument, an integer N, and returns the [factorial](https://en.wikipedia.org/wiki/Factorial) for that number. The factorial is the multiplication of all integers from 1 to N and has the special case that the factorial of 0 is 1. This method only accepts 0 and positive integers, so if a negative integer is given it should raise an exception.
- Create a method called reverse that takes one argument, a string word, and returns word reversed (e.g. if word is "hello" it returns "olleh").
- Create a method called fizzbuzz that takes one argument, an integer N, and returns a string. The returned string is constructed following these rules:
When N is divisible by 3, return "fizz".
When N is divisible by 5, return "buzz".
When N is divisible by 3 and 5, return "fizzbuzz".
In any other case, return N as a string (e.g. say N is 7 then return "7").
- Remember to start with tests for all of the methods described above. Your commit history should prove that you have used TDD.

### CODE REVIEW

File: spec/solver_spec.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+describe Solver do
+  before :each do
+    @solver = Solver.new
+  end
+
+  context 'To test the solver class' do
+    it 'Returns the factorial of the number' do
+      expect(@solver.factorial(5)).to eq 120
+    end
```
Comment (omarsalem7) [inline spec/solver_spec.rb]: - [x] Kindly add a test case that handles negative values which raise an exception. Also, consider adding a test case when input is 0, the result should equal 1.

File: spec/solver_spec.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+    it 'Reverses the String input' do
+      expect(@solver.reverse('abby')).to eq 'ybba'
+    end
+
+    it 'Runs the fizzbuzz algorithm' do
+      expect(@solver.fizzbuzz(15)).to eq 'FizzBuzz'
+    end
```
Comment (omarsalem7) [inline spec/solver_spec.rb]: - [x] Also consider adding test cases that handle numbers divisible by 3 and 5, also for numbers not divisible by 3 or 5 like 7

**Review by omarsalem7 — CHANGES_REQUESTED** [review]
Hi Team 😊, 

Good job so far! 👏 
You have done well reaching this point of the milestone, but there are some issues that you still need to work on to go to the next project but you are almost there! 

you-are-near

## Highlights
- There is no linter error present ✔️
- The Pull Request has a proper title and description. ✔️
- Correct git-flow ✔️

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/solver_spec.rb (hunk 2 above)
Comment (AbbyNyakara) [reply spec/solver_spec.rb]: Added the additional test cases

File: spec/solver_spec.rb (hunk 1 above)
Comment (AbbyNyakara) [reply spec/solver_spec.rb]: I have added a test that raises an argument error when the number is negative

**Review by FrederickMih — APPROVED** [review]
Hi @gasangw,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- Correct GitHub workflow :heavy_check_mark: 
- PR has a title and summary :heavy_check_mark: 
- No Linter error :heavy_check_mark: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/solver_spec.rb (hunk 1 above)
Comment (AbbyNyakara) [reply spec/solver_spec.rb]: @omarsalem7 , i added the test for a negative number, but when the test runs it raises an argument error, which is correct as per my code. However the test fails because its an error raised. So essentially, has the test passed?. Please refer to the screenshots
image
image
